Lib/fontParts/base/point.py

```diff
@@ -1,15 +1,18 @@
-from fontTools.misc import transform
```

Comment:
I would import annotations from`__future__` as a general principle and keep annotation related imports at the top. Also, I think it's a good idea to keep any imports only used for annotation within a `TYPE_CHECKING` conditional. Lastly, I like the one-import-per-line scheme used originally, but that may just be me:
```python
from __future__ import annotations
from typing import TYPE_CHECKING, Any, Optional, Union, Tuple
from fontTools.misc import transform
from fontParts.base.base import (
    BaseObject,
    TransformationMixin,
    PointPositionMixin,
    SelectionMixin,
    IdentifierMixin,
    dynamicProperty,
    reference
)
from fontParts.base import normalizers
from fontParts.base.deprecated import DeprecatedPoint, RemovedPoint
from fontParts.base.annotations import(
    IntFloatType,
    TransformationMatrixType
)

if TYPE_CHECKING:
    from fontParts.base.font import BaseFont
    from fontParts.base.glyph import BaseGlyph
    from fontParts.base.lib import BaseLib
    from fontParts.base.contour import BaseContour
```

Reply:
Agreed!